src/qtism/runtime/storage/binary/QtiBinaryStreamAccess.php

```diff
@@ -1187,7 +1189,9 @@ public function readFile()
             return FileHash::createFromArray($decoded);
         }
 
-        return $this->getFileManager()->retrieve($id);
+        [$id, $filename] = sscanf($id, self::FILE_IDENTIFIER_ENCODING_MASK);
```

Comment:
Neat, I love it!

Now, if possible, could you please add two test cases?
1. checking the new marshaling/unmarshaling
2. checking the old format (just the $id) unmarhsaling